File: apps/push-service/src/main/java/com/example/chat/push/infrastructure/kafka/PushResultProducer.java
package com.example.chat.push.infrastructure.kafka;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

/**
 * 푸시 처리 결과 Kafka 발행자
 */
@Component
@RequiredArgsConstructor
@Slf4j
public class PushResultProducer {
    private static final String TOPIC = "push-result-events";

    private final KafkaTemplate<String, String> kafkaTemplate;
    private final ObjectMapper objectMapper;

    public void sendResult(PushResultEvent result) {
        try {
            String message = objectMapper.writeValueAsString(result);
            log.info("Publishing push result: {}", message);
            kafkaTemplate.send(TOPIC, message);
        } catch (Exception e) {
            log.error("Failed to publish push result for messageId={}", result.pushMessageId(), e);
        }
    }
}

File: apps/push-service/src/main/kotlin/com/example/chat/push/infrastructure/kafka/PushResultProducer.kt
package com.example.chat.push.infrastructure.kafka

import com.fasterxml.jackson.databind.ObjectMapper
import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component

private val logger = KotlinLogging.logger {}

@Component
class PushResultProducer(
    private val kafkaTemplate: KafkaTemplate<String, String>,
    private val objectMapper: ObjectMapper
) {

    fun sendResult(result: PushResultEvent) {
        val topic = "push-result-events"
        val message = objectMapper.writeValueAsString(result)
        logger.info { "Publishing push result: $message" }
        kafkaTemplate.send(topic, message)
    }
}

data class PushResultEvent(
    val pushMessageId: Long,
    val targetUserId: String,
    val status: String,
    val errorMessage: String? = null
)
